libs/lb/lb-rs/src/lb_client.rs

```diff
@@ -0,0 +1,620 @@
+            .await
+            .map_err(core_err_unexpected)?;
+        let args = bincode::serialize(&(path.to_string(), target_id))
+            .map_err(core_err_unexpected)?;
+        call_rpc(&mut stream, "create_link_at_path", Some(args)).await
+    }
+
+    pub async fn create_at_path(&self, path: &str) -> LbResult<File>{
+        let mut stream = TcpStream::connect(&self.addr)
+            .await
+            .map_err(core_err_unexpected)?;
+        let args = bincode::serialize(&path.to_string())
+            .map_err(core_err_unexpected)?;
+        call_rpc(&mut stream, "create_at_path", Some(args)).await
+    }
+
+    pub async fn get_by_path(&self, path: &str) -> LbResult<File>{
+        let mut stream = TcpStream::connect(&self.addr)
+            .await
+            .map_err(core_err_unexpected)?;
+        let args = bincode::serialize(&path.to_string())
+            .map_err(core_err_unexpected)?;
+        call_rpc(&mut stream, "get_by_path", Some(args)).await
+    }
+
+    pub async fn get_path_by_id(&self, id: Uuid) -> LbResult<String>{
+        let mut stream = TcpStream::connect(&self.addr)
+            .await
+            .map_err(core_err_unexpected)?;
+        let args = bincode::serialize(&id)
+            .map_err(core_err_unexpected)?;
+        call_rpc(&mut stream, "get_path_by_id", Some(args)).await
+    }
+
+    pub async fn list_paths(&self, filter: Option<Filter>) -> LbResult<Vec<String>>{
+        let mut stream = TcpStream::connect(&self.addr)
+            .await
+            .map_err(core_err_unexpected)?;
+        let args = bincode::serialize(&filter)
+            .map_err(core_err_unexpected)?;
+        call_rpc(&mut stream, "list_paths", Some(args)).await
+    }
+
+    pub async fn list_paths_with_ids(
+        &self, filter: Option<Filter>,
+    ) -> LbResult<Vec<(Uuid, String)>>{
+       let mut stream = TcpStream::connect(&self.addr)
+            .await
+            .map_err(core_err_unexpected)?;
+        let args = bincode::serialize(&filter)
+            .map_err(core_err_unexpected)?;
+        call_rpc(&mut stream, "list_paths_with_ids", Some(args)).await
+    }
+
+    pub async fn share_file(&self, id: Uuid, username: &str, mode: ShareMode) -> LbResult<()>{
+        let mut stream = TcpStream::connect(&self.addr)
+            .await
+            .map_err(core_err_unexpected)?;
+        let args = bincode::serialize(&(id,username.to_string(),mode))
+            .map_err(core_err_unexpected)?;
+        call_rpc(&mut stream, "share_file", Some(args)).await
+    }
+
+    pub async fn get_pending_shares(&self) -> LbResult<Vec<File>>{
+        let mut stream = TcpStream::connect(&self.addr)
+            .await
+            .map_err(core_err_unexpected)?;
+        call_rpc(&mut stream, "get_pending_shares", None).await
+    }
+
+    pub async fn reject_share(&self, id: &Uuid) -> Result<(), LbErr>{
+        let mut stream = TcpStream::connect(&self.addr)
```

Comment:
This identical connection can be moved to `call_rpc`, which will further reduce the amount of duplication and make it easier to do something more sophisticated later.